content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,266 @@
+
+```{r eval=FALSE}
+?usethis::git_vaccinate() # read the help file
+usethis::git_vaccinate()
+?usethis::create_github_token() # read the help file
+usethis::create_github_token() #browser opens, follow instructions
+```
+
+Add the PAT to the Git Credential Manager:
+
+```{r eval=FALSE}
+?gitcreds::gitcreds_set() # read the help file
+gitcreds::gitcreds_set() #paste PAT
+```
+
+**WARNING**:  handle your Personal Access Token (PAT) as as secret.
+Anyone who has your token, has access to your GitHub account.
+
+Check if everything is OK:
+
+```{r eval=FALSE}
+usethis::git_sitrep()
+```
+
+The default Git protocol should be 'https' (recommended on Windows).
+
+## Modes of authentication
+
+The way in which you need to authenticate depends on how you access your resources in GitHub.
+For instance, to manage your repositories online you can sign-in to GitHub using a username and password followed by two-factor authentication.
+As another example, a function like `checklist::new_branch()` will create a new branch on GitHub and locally.
+In order to create a new branch, you will need a Personal Access Token with appropriate scopes if you use the HTTPS protocol.
+A token with no assigned scopes can only access public information.
+
+We will follow the [recommendations given in the usethis package](https://usethis.r-lib.org/articles/articles/git-credentials.html) for Git authentication on Windows, which covers all ways which you will need to safely authenticate:
+
+1. Turn on two-factor authentication for your GitHub account.
+1. Adopt HTTPS as your Git transport protocol.
+1. Use a personal access token (PAT) for all Git remote operations from the command line or from R.
+1. Allow tools to store and retrieve your credentials from the Git Credential Manager.
+
+This deserves some further explanation.
+The first recommendation essentially is an extra layer of security compared to a simple username and password only authentication.
+To enable two-factor authentication for your GitHub account, follow [these steps](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#configuring-two-factor-authentication-using-a-totp-mobile-app).
+Another two-factor authentication app that is not mentioned on that website but easy to use is Google Authenticator.
+
+The second recommendation is to use the [HTTPS protocol](#https) to transport information (data) from your local Git repositories to their remote (online) counterparts.
+A good alternatives is the SSH protocol. Do not use the HTTP protocol as it is not secure.
+We have also included in this tutorial how to use the [SSH protocol](#ssh), but in our experience this is not necessary in most circumstances and more involved to implement in Windows.
```

Comment:
The `#ssh` fragment is non-functional.

Reply:
See [918729b](https://github.com/inbo/tutorials/pull/309/commits/918729b1fba56ce55c446cd00bea10b97d325aed)